presto-native-execution/presto_cpp/main/types/PrestoToVeloxQueryPlan.cpp

```diff
@@ -989,7 +989,11 @@ VeloxQueryPlanConverterBase::toVeloxQueryPlan(
   bool streamable = !node->preGroupedVariables.empty() &&
       node->groupingSets.groupingSetCount == 1 &&
       node->groupingSets.globalGroupingSets.empty();
-
+  // TODO karteekmurthys@ Re-enable after
+  // fixing:https://github.com/prestodb/presto/issues/22585
+  if (streamable) {
```

Comment:
sorry for all the back and forth. instead of failing in plan conversion for aggregation as you are doing here, can we fail in plan conversion for join if it's a nested loop join? That would be a more robust mitigation because it's the root cause of the issue, and there are other optimizations aside from streaming aggregation that might be assuming nested loop join is preserving order when it's not e.g. we might remove an order by because we think the input is already ordered or plan a merge join because we think the input is ordered.

Reply:
oh looks like https://github.com/prestodb/presto/pull/23341/files is already doing this.

Reply:
FYI I am turning off NLJ here : https://github.com/prestodb/presto/pull/23341 .